### ISSUE TITLE
kubelet may detect GCE when running kind on GCE

### ISSUE THREAD

**Issue by BenTheElder** [kubernetes-sigs/kind#426]
**What happened**: kubelet appears to have read `/sys/class/dmi/id/product_name` when running kind on GCE and detected "Google Compute Engine", it appears to fail if it can't talk to the metadata server.

**What you expected to happen**: Kubernetes should not "be in cloud mode" so to speak.

**How to reproduce it (as minimally and precisely as possible)**: run a kind cluster on GCE

**Anything else we need to know?**:

**Environment:**

- kind version: (use `kind version`): 0.2.1
- Kubernetes version: (use `kubectl version`): 1.13.4
- Docker version: (use `docker info`): 
- OS (e.g. from `/etc/os-release`):

**Comment by BenTheElder** [comment]
/assign

we should probably try to properly set cloud provider to none by default AND/OR always mount a special product_name by default.

current workaround:
```bash
  echo "kind" >./product_name
  cat <<EOF >"./kind-config.yaml"
kind: Cluster
apiVersion: kind.sigs.k8s.io/v1alpha3
nodes:
- role: control-plane
  extraMounts:
  - containerPath: /sys/class/dmi/id/product_name
    hostPath: ${PWD}/product_name
EOF
```

**Comment by BenTheElder** [comment]
This seems to be one of the blockers for https://github.com/kubernetes-sigs/kind/issues/182.

**Comment by aojea** [comment]
`kind` remounts the `/sys` folder in the nodes as ro https://github.com/kubernetes-sigs/kind/blob/161151a26faf0dbe962ac9f323cc0cdebac79ba8/pkg/cluster/nodes/node.go#L221 thus we can´t rewrite the file.

However, seems that kubernetes only uses the `/sys/class/dmi/id/product_name` to check if is running on a GCE VM
https://github.com/kubernetes/kubernetes/blob/716344fd7d68f92d9a16280be1feae0ffe122bb4/pkg/credentialprovider/gcp/metadata.go#L47 when using the GCP credential provider.

@BenTheElder  I´m not familiar with the credential provider feature, is it possible to disable it?

**Comment by aojea** [comment]
Ben´s idea works 
 `echo kind > /kind/product_name && mount -o ro,bind /kind/product_name /sys/class/dmi/id/product_name`

**Comment by BenTheElder** [comment]
fixed by #461
